backend/contracts/ElectionFactory.sol

```diff
@@ -1,28 +1,51 @@
-//SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: UNLICENSED
 
 pragma solidity ^0.8.0;
 
 contract ElectionFactory {
-    struct Election {
+    struct Candidate {
+        uint8 id;
         string name;
     }
 
-    Election[] public elections;
-
-    constructor() {
+    struct Ballot {
+        string name;
+        string description;
+        uint256 createTime;
+        uint256 votingTime;
+        Candidate[] candidates;
     }
 
-    function createElection(string memory _name) external{
-        Election memory newElection = Election(_name);
-        elections.push(newElection);
+    mapping(uint256 => Ballot) public ballots;
+    uint256 public nextBallotId;
+
+    event CandidateAdded(uint256 ballotId, uint8 id, string name);
+    event ContractInitialized(uint256 ballotId, string name, string description, uint256 createTime, uint256 votingTime, Candidate[] candidates);
+
+    function clearCandidates(uint256 _ballotId) internal {
+        delete ballots[_ballotId].candidates; // just a precaution for any random data that might have left in array
     }
 
-    function getAllElectionNames() external view returns (string[] memory) {
-        uint256 length = elections.length;
-        string[] memory names = new string[](length);
-        for (uint256 i = 0; i < length; i++) {
-            names[i] = elections[i].name;
+    function initialize(string memory _name, string memory _description, uint256 _votingTime, string[] memory _candidateNames) public {
+
+        // check for a number of elements in given array (min. 2 candidates, max. - 10)
+        require(_candidateNames.length >= 2 && _candidateNames.length <= 10, "Invalid number of candidates");
+
+        uint256 ballotId = nextBallotId++;
+        Ballot storage newBallot = ballots[ballotId];
+
+        newBallot.name = _name;
+        newBallot.description = _description;
+        newBallot.createTime = block.timestamp;
+        newBallot.votingTime = _votingTime; // takes the value of the voting lifetime in seconds
+
+        clearCandidates(ballotId);
+
+        for(uint i = 0; i < _candidateNames.length; i++) { // takes array of strings (candidate names) and converts it to array of structs (id + name)
+            newBallot.candidates.push(Candidate(uint8(i), _candidateNames[i]));
+            emit CandidateAdded(ballotId, uint8(i), _candidateNames[i]);
         }
-        return names;
+
+        emit ContractInitialized(ballotId, _name, _description, newBallot.createTime, _votingTime, newBallot.candidates);
```

Comment:
This event name also seems to have misleading name, considered naming it something like `BallotCreated`